tests/shakedown/shakedown/clients/node.py

```diff
@@ -0,0 +1,132 @@
+from abc import ABC, abstractmethod
+import logging
+import os
+import requests
+import urllib
+
+from . import dcos_url_path
+from .authentication import dcos_acs_token, DCOSAcsAuth
+
+logger = logging.getLogger(__name__)
+
+class Collection(ABC):
+    """Resources have collections of objects."""
+
+    @abstractmethod
+    def create(self):
+        pass
+
+class Session(ABC):
+
+    def create_url(self, path):
+        """Create the URL based off this partial path."""
+        return urllib.parse.urljoin(self.base_url, path)
+
+    def get(self, path, *args, **kwargs):
+        url = self.create_url(path)
+        kwargs['auth'] = self.auth
+        return requests.get(url, *args, **kwargs)
+
+    def post(self, path, *args, **kwargs):
+        url = self.create_url(path)
+        kwargs['auth'] = self.auth
+        return requests.post(url, *args, **kwargs)
+
+
+class DiagnosticBundle():
+
+    def __init__(self, session, bundle_name):
+        logger.info('Created diagnostic bundle %s', bundle_name)
+
+        self.session = session
+        self.bundle_name = bundle_name
+
+    def status(self):
+        """Check the status of the bundle creation.
+
+        :returns None if the bundle is not done yet, the bundle's file name otherwise.
+        """
+        logger.info('Retrieve status for %s', self.bundle_name)
+        resp = self.session.get('list/all')
+        resp.raise_for_status()
+
+        nodes = resp.json()
+
+        # Flatten bundle lists
+        bundle_list = [bundle for bundle_list in nodes.values() if bundle_list is not None for bundle in bundle_list]
+
+        # Search for our bundle.
+        for bundle in bundle_list:
+            file_name = bundle['file_name']
+            if os.path.basename(file_name) == self.bundle_name:
+                return file_name
+
+        return None
+
+    def download(self, bundle_path):
+        file_name = self.status()
+        bundle_name = os.path.basename(file_name)
+        assert file_name is not None, 'The bundle is not ready yet.'
```

Comment:
I assume that the bundle is done once it's part of the list. Is this assumption correct, @janisz?